### PR TITLE
Final Program Group reassignments.

### PR DESCRIPTION
**Pull request by cmnbroad**
This removes the remaining obsolete program groups and reassigns the remaining tools.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/spark/PileupSpark.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
 import org.broadinstitute.hellbender.engine.filters.WellformedReadFilter;
 import org.broadinstitute.hellbender.engine.spark.LocusWalkerContext;
+import org.broadinstitute.hellbender.tools.walkers.annotator.Coverage;
```
Comment (cmnbroad) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/PileupSpark.java]: Remove this.

File: src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java [hunk 2]
```diff
@@ -28,13 +28,13 @@
 import java.util.*;
 import java.util.stream.Collectors;
 
-@BetaFeature
+@ExperimentalFeature
```
Comment (cmnbroad) [inline src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java]: Should this be a `@DocumentedFeature` ?

File: src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java [hunk 3]
```diff
@@ -28,13 +28,13 @@
...
 @CommandLineProgramProperties(summary = "Fix the sample names in a vcf that ran through the GenomicsDBImport when " +
         "the batch ordering bug was present, this will restore the correct sample names provided it is given the exact sample name mapping / vcf ordering" +
         "and batch sized that was used in the initial import. " +
         "See https://github.com/broadinstitute/gatk/issues/3682 for more information",
         oneLineSummary = "fix sample names in a shuffled callset",
-        programGroup = VariantProgramGroup.class,
+        programGroup = ShortVariantDiscoveryProgramGroup.class,
```
Comment (cmnbroad) [inline src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java]: Is this correct ? It really only matters if we expose it to doc/command line per above.

File: src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java (hunk 2 above)
Comment (sooheelee) [reply src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java]: Yes, it should be.

File: src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java (hunk 3 above)
Comment (sooheelee) [reply src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java]: My vote goes for Other program group.

**Comment by sooheelee** [comment]
I checked only the tools that are marked `no` in the `correct category in gatk --list` in the 0107Check_category_&_doc tab of https://docs.google.com/spreadsheets/d/19SvP6DHyXewm8Cd47WsM3NUku_czP2rkh4L_6fd-Nac/edit?usp=sharing.

## The following tools need fixing:

tool | category   philosophically ok? | 2nd   check, correct category in gatkDoc? | categorization   fixed by https://github.com/broadinstitute/gatk/pull/4094?
-- | -- | -- | --
IndexFeatureFile | NO | y | no; still in Variant Manipulation but should be in Other.
CreateHadoopBamSplittingIndex | NO | y | no; still in Other but no longer with   ConvertHeaderlessHadoopBamShardToBam
VariantAnnotator | y | DOES NOT SHOW UP | no;   does not show up
FixCallsetSampleOrdering | MISSING | no | no;   does not show up
DepthOfCoverage | y | DOES NOT SHOW UP | no
DiagnoseTargets | y | DOES NOT SHOW UP | no
GatherTranches | y | y | DOES NOT SHOW UP IN GATKDOC; shows up in correct category in gatk --list
ValidateBasicSomaticShortMutations | MISSING | no | DOES NOT SHOW UP IN GATKDOC; shows up in correct category in gatk --list

## The following tools appear fixed by this PR:

tool | category   philosophically ok? | 2nd   check, correct category in gatkDoc? | categorization   fixed by https://github.com/broadinstitute/gatk/pull/4094?
-- | -- | -- | --
CollectBaseDistributionByCycleSpark | y | y | y
ASEReadCounter | y | y | y
CountBases | y | y | y
CountBasesSpark | y | y | y
CountReads | y | y | y
CountReadsSpark | y | y | y
PileupSpark | y | y | y
CollectInsertSizeMetricsSpark | y | y | y
CollectMultipleMetricsSpark | y | y | y
CollectQualityYieldMetricsSpark | y | y | y
CompareBaseQualities | y | y | y
EstimateLibraryComplexityGATK | y | y | y
FilterLongReadAlignmentsSAMSpark | y | y | y
FlagStat | y | y | y
FlagStatSpark | y | y | y
MeanQualityByCycleSpark | y | y | y
QualityScoreDistributionSpark | y | y | y
GatherBQSRReports | y | y | y
ApplyBQSR | y | y | y
ApplyBQSRSpark | y | y | y
BaseRecalibrator | y | y | y
BaseRecalibratorSpark | y | y | y
BaseRecalibratorSparkSharded | y | y | y
BQSRPipelineSpark | y | y | y
BwaSpark | y | y | y
ClipReads | y | y | y
FixMisencodedBaseQualityReads | y | y | y
LeftAlignIndels | y | y | y
MarkDuplicatesGATK | y | y | y
MarkDuplicatesSpark | y | y | y
PrintReads | y | y | y
PrintReadsSpark | y | y | y
RevertBaseQualityScores | y | y | y
SortReadFileSpark | y | y | y
SplitNCigarReads | y | y | y
SplitReads | y | y | y
UnmarkDuplicates | y | y | y
CombineGVCFs | y | y | y
GenomicsDBImport | y | y | y
GenotypeGVCFs | y | y | y
HaplotypeCaller | y | y | y
HaplotypeCallerSpark | y | y | y
AnnotatePairOrientation | y | y | y
AnnotateVcfWithBamDepth | y | y | y
AnnotateVcfWithExpectedAlleleFraction | y | y | y
CalculateGenotypePosteriors | y | y | y
CalculateMixingFractions | y | y | y
Concordance | y | y | y
CountFalsePositives | y | y | y
CountVariantsSpark | y | y | y
Funcotator | y | y | y
ApplyVQSR | y | y | y
VariantFiltration | y | y | y
VariantRecalibrator | y | y | y
SelectVariants | y | y | y

**Comment by sooheelee** [comment]
Back to you @cmnbroad

**Comment by cmnbroad** [comment]
Thank @sooheelee. DepthOfCoverage and DiagnoseTargets don'e exist. I'll fix the others.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/PileupSpark.java (hunk 1 above)
Comment (cmnbroad) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/PileupSpark.java]: Done.

File: src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java (hunk 2 above)
Comment (cmnbroad) [reply src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java]: Done.

File: src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java (hunk 3 above)
Comment (cmnbroad) [reply src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java]: Done.

**Comment by cmnbroad** [comment]
Ok, all done. I had to rebase again too. Pushed, tests running.

**Pull request merged**